docs/adr/0001-extend-tosca-yaml-model.md

```diff
@@ -0,0 +1,45 @@
+# Extend TOSCA YAML model to a full in-memory model
+
+As a programmer, I want to access the complete YAML model and not be forced to traverse through all files.
+
+For instance, `TNodeTemplate.java` offers `public QName getType() {`, but one wants `public TNodeType getType()`.
+
+## Considered Alternatives
+
+* Inherit from plain data model and add transparent resolution
+* New data model
+* Eclipse EMF
+
+## Decision Outcome
+
+* Chosen Alternative: Inherit from plain data model and add transparent resolution
+* Comes out best (see below)
+
+
+## Pros and Cons of the Alternatives <!-- optional -->
+
+### Inherit from plain data model and add transparent resolution
+
+A new model with all classes in a different package inheriting from all model classes.
+Each class gets new methods.
+For the type example:
+
+- `public TNodeType getTypeResolved()`
+- `public setType(TNodeType nodeType)`
+
+This model is offered by a new repository implementation, which resolves the objects on demand.
```

Comment:
I think this won't work (or I don't understand what you want to do) nor is it neccessary. 
Why do we need a repository? All yaml template files get parsed, merged into one TServiceTemplate, and then get resolved in-memory. Anything I miss?

Reply:
The repository implementation can rely on files stored in a file system. Just a simple repository which parses ALL the files and returns the necessary objects.